packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-train-task.ts

```diff
@@ -105,6 +105,11 @@ export class SagemakerTrainTask implements ec2.IConnectable, sfn.IStepFunctionsT
      */
     private readonly stoppingCondition: StoppingCondition;
 
+    private readonly vpc: ec2.IVpc;
+    private readonly securityGroup: ec2.ISecurityGroup;
+    private readonly securityGroups: ec2.ISecurityGroup[] = [];
+    private readonly subnets: string[];
+
     constructor(scope: Construct, private readonly props: SagemakerTrainProps) {
```

Comment:
If you need a `scope`, we usually defer all the work to the `bind()` function which will have a scope.

Reply:
Removed scope from constructor of both SageMaker task objects. Now creates Role and SecurityGroups in the `bind()` method